enterprise/backend/src/metabase_enterprise/sandbox/query_processor/middleware/row_level_restrictions.clj

```diff
@@ -173,12 +174,13 @@
 
 (mu/defn- native-query-metadata :- [:+ :map]
   [source-query :- [:map [:source-query :any]]]
-  (let [result (mw.session/as-admin
-                 ((requiring-resolve 'metabase.query-processor/process-query)
-                  {:database (u/the-id (lib.metadata/database (qp.store/metadata-provider)))
-                   :type     :query
-                   :query    {:source-query source-query
-                              :limit        0}}))]
+  (let [result (binding [qp.pipeline/*result* qp.pipeline/default-result-handler]
```

Comment:
I think we should have a comment explaining why this binding occurs, since it's super not obvious. Nice job finding this problem btw!